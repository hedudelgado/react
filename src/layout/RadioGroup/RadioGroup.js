import React from 'react';
import { array } from 'prop-types';

const RadioGroup = ({ children }) => (
  <div className="wb-form-item fixed-size mb3 wb-form-item--clean max-fit">
    <ul className="wb-radio__group">
      {children}
    </ul>
  </div>
);

RadioGroup.propTypes = {
  children: array,
};

export default RadioGroup;
