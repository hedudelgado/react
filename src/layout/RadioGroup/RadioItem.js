import React from 'react';
import {
  bool,
  string,
  array,
  object,
  oneOfType,
  number,
} from 'prop-types';
import classnames from 'classnames';

const RadioItem = ({
  label,
  subLabels,
  field: {
    name,
    onBlur,
    onChange,
    value: fieldValue,
  },
  value,
  id,
  sessioncamhide,
  ...props
}) => {
  const labels = subLabels.map(subLabel => <span key={subLabel} className="wb-radio__label__inner-info">{subLabel}</span>);
  const classes = classnames({
    sessioncamhidetext: sessioncamhide,
  });
  return (
    <li className="wb-radio__inner">
      <label htmlFor={id} className="wb-radio__label">
        <input
          type="radio"
          name={name}
          id={id}
          onBlur={onBlur}
          onChange={onChange}
          checked={value === fieldValue}
          value={value}
          className={classes}
          {...props}
        />
        <span className="wb-radio__label__inner">
          <span className="wb-radio__label__inner__title">{label}</span>
          {labels}
        </span>
      </label>
    </li>
  );
};

RadioItem.propTypes = {
  value: oneOfType([
    string,
    number,
  ]),
  label: oneOfType([
    string,
    number,
  ]),
  subLabels: array,
  field: object,
  id: oneOfType([
    string,
    number,
  ]),
  /**
    Sessioncam field masking.
  */
  sessioncamhide: bool,
};

RadioItem.defaultProps = {
  subLabels: [],
};

export default RadioItem;
