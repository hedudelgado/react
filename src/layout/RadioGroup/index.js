export { default } from './RadioGroup';

export { default as RadioItem } from './RadioItem';
