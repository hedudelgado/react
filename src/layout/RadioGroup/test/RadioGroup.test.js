import React from 'react';
import { shallow } from 'enzyme';
import RadioGroup from '../RadioGroup';
import RadioItem from '../RadioItem';

describe('RadioGroup Component', () => {
  const children = [
    <RadioItem id="first" key="true" label="test label" name="test" value="true" />,
    <RadioItem id="second" key="false" label="test label" name="test" value="false" />,
  ];
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <RadioGroup>
        {children}
      </RadioGroup>,
    );
  });

  describe('Renders RadioGroup Element', () => {
    it('should render the component', () => {
      expect(wrapper).not.toBeEmptyRender();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
