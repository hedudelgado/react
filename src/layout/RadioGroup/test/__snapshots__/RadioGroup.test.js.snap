// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Component Renders RadioGroup Element should render the component 1`] = `
<div
  className="wb-form-item fixed-size mb3 wb-form-item--clean max-fit"
>
  <ul
    className="wb-radio__group"
  >
    <RadioItem
      id="first"
      key="true"
      label="test label"
      name="test"
      subLabels={Array []}
      value="true"
    />
    <RadioItem
      id="second"
      key="false"
      label="test label"
      name="test"
      subLabels={Array []}
      value="false"
    />
  </ul>
</div>
`;
